### ISSUE TITLE
[BUG] Currently pressed key is not removed if shift is released first

### ISSUE THREAD

**Issue by damusnet** [JohannesKlauss/react-hotkeys-hook#1064]
**Describe the bug**
The order in which combo keys are released matters, when it should not.

**To Reproduce**
1. Setup a `shift+?` hotkey
2. Press `shift`, press `?`, release `shift` first, release `?` second
3. Press `shift` again
4. Hotkey triggers, even though `?` is not pressed

[CodeSanbdox Repro](https://codesandbox.io/p/sandbox/gifted-black-fr7cxq?file=%2Fapp%2Fpage.tsx%3A9%2C6)

**Expected behavior**
Hotkey should not trigger again on `shift` only, since `?` is not pressed anymore

**Screenshots**
<img width="1840" alt="image" src="https://github.com/JohannesKlauss/react-hotkeys-hook/assets/433409/84c33a24-3ff1-4b3a-bdf7-190dd129a86b">

**Comment by damusnet** [comment]
After a bit of investigation, this sort of makes sense. When pressing combo keys, the following is (roughly) registered:
* Add `shift`
* Add `?`
* Remove `shift`
* Remove `/` <= that's the issue ; it's because without the `shift`, the `?` key is now registered as a `/` key. So `?` is never removed from the currently pressed keys, and any press to `shift` triggers the combo again.

This is on a en-US keyboard, but I suspect it's true for any `shift` combo on any keyboard. The particular key here is:
```
┌─────┐
│  ?  │
│  /  │
└─────┘
```

Not sure how best to fix this, as I don't fully understand the entire implementation logic.

**Comment by tamerxkilinc** [comment]
I solved this by doing the same as for the command key. On macOS the keyup event of letters and digits is not triggering while holding the command key down. That's why you can't just remove single keys of the array but need to clear it. The way I solved this also worked for the shift key!

Set a timeout each time shift is pressed. When it is released, instead of trying to remove the specific characters which were pressed with the shift key, the array of pressed keys is cleared completely. This ensures that you have no characters left in the pressed keys array.

**Comment by JohannesKlauss** [comment]
Yes this is currently a problem, because the library mixes the produces symbol with the produced key code. This will be fixed in version 5.

**Comment by JohannesKlauss** [comment]
This is a layout specific problem that gets addressed in version 5.

This has been fixed in pre release [5.0.0-1](https://www.npmjs.com/package/react-hotkeys-hook/v/5.0.0-1). Please try out this version.
To listen to `=` use you just use `useHotkeys('=', callback, {useKey: true})`

Please let me know if that fixes your issue.

**Comment by TomPridham** [comment]
It seems like the property should prolly be `useCode: true`, right? For things like `.`, the key changes when shift is pressed, but the code stays the same

**Comment by TomPridham** [comment]
I'd be happy to update your changes from the 5.0.0-1 branch if that sounds right to you

**Comment by JohannesKlauss** [comment]
This is fixed in 5.0.1
